### ISSUE TITLE
Increasing the number of accepted CrosRef citations

### ISSUE THREAD

**Issue by HughP** [zuphilip/zotero-open-citations#16]
It seems from my limited experimentation that I one asks the OCIC API for citations, that an unlimited number is returned and written into a note. However, limits are imposed on the import from CrossRef on these DOIs returned from OCIC. These limits are set in the preference pane.  

I ran the OCIC query on a resource in Zotero with the default value of 3. three resources showed up in my folder on the left side.
I ran the OCIC query on a resource in Zotero with a custom value of 7. the three original resources showed up in a new folder with an additional 4 resources. 

I think it makes sense to do two things:
1. don't create a new folder if it already exists.
2. don't download the citations again, because every time the plugin is run on a batch of resources it will import the previously imported ones again.

**Comment by zuphilip** [comment]
I agree on the first point, and the second one also makes sense. If someone wants to force reloading all referencing from Crossref, then it is still possible to delete the complete folder and run the plugin again.

What should happen if you run the plugin again? Just stop because we already have three items in the citations folder? I think we could also try to receive another three items then into the folder, such that after two rounds of clicking on the plugin, there are 6 citations, etc. What do you think?

**Comment by HughP** [comment]
>What should happen if you run the plugin again?

the same folder as the first call should be used (or at least attempted) 
- if there are three already in the folder and "3" is the number set in the OCIC setting then ask the user if they intended to download more if they are available, and if they intended to download more then please up the setting in the preferences.

- If the setting is 7 and we have three in the folder then assume that the OCIC setting has been raised previously and we are only adding new ones to the folder, but they should not be "new additions to Zotero" rather if there is a match to what is in Zotero already then that reference should be added to the folder, rather than creating a new record in Zotero.... the principle should be create as few new records in Zotero as possible. 

> I think we could also try to receive another three items then into the folder, such that after two rounds of clicking on the plugin, there are 6 citations, etc. What do you think?

This behavior choice should be made clear in the preference panel. I kind of don't want to keep asking the plugin to fetch more... I just want the complete set of results all at once... but I also want the API to like me as a user... so what are the api limits?

**Comment by zuphilip** [comment]
@HughP Please try out the new version 0.3.0. The first point is implemented and it will be made sure that always the latest N citations are in the subcollection.

I haven't implemented any ineractive dialog with the user, because this could lead to hundreds or thousands of such dialogs when you select that many items.

(I don't know whether there are rate limits for the COCI or Crossref service. However, from your report I understood that the bottleneck might be Zotero or the Plugin itself.)

**Comment by HughP** [comment]
> I haven't implemented any ineractive dialog with the user, because this could lead to hundreds or thousands of such dialogs when you select that many items.

Usually there is a check-box for "apply all to following" ... like when you replace a file in a finder...

**Comment by zuphilip** [comment]
Spinning wheels or a progress bar could help the user to see better how far the process is. However, I don't have any experience there and I don't want to spend too much time in UX in general. Further discussion in #18. Maybe we can at least give a short notice after the whole process is done.

We can check with the people behind COCI, if they are okay with the current calls from the plugin.

> I tried version 0.3 on a resource that had previously been cataloged as having 352 citations from COCI, but only three were added to the folder, when I updated the default setting to 9 still only three were in the folder.

I tested this for myself again it worked fine. Restart Zotero and try again, wait long enough, click on another collection and then back again etc. If the problem persist, then I need more information, ideally an example I can replicate step by step.
